mm2src/coins/lp_coins.rs

```diff
@@ -3555,6 +3638,23 @@ pub async fn verify_message(ctx: MmArc, req: VerificationRequest) -> Verificatio
     Ok(VerificationResponse { is_valid })
 }
 
+pub async fn sign_raw_transaction(ctx: MmArc, req: SignRawTransactionRequest) -> SignRawTransactionResult {
+    let coin = lp_coinfind_or_err(&ctx, &req.coin).await?;
+    coin.sign_raw_tx(&req).await
+}
+
+pub async fn sign_eth_transaction(ctx: MmArc, req: SignEthTransactionRequest) -> SignEthTransactionResult {
+    let coin = lp_coinfind_or_err(&ctx, &req.coin).await?;
+    match coin {
+        MmCoinEnum::EthCoin(eth_coin) => sign_eth_tx(&eth_coin, &req).await,
+        _ => {
+            return MmError::err(RawTransactionError::NotImplemented {
+                coin: coin.ticker().to_string(),
+            })
+        },
+    }
+}
```

Comment:
Again, these should be only one function and one RPC. For the request, you can use an enum or any other approach you see fit. e.g.
```rust
struct SignTransactionRequest {
    coin: String,
    #[serde(flatten)]
    tx: SignTransactionEnum,
}

#[serde(tag = "type", content = "tx")]
enum SignTransactionEnum {
    Utxo(SignRawTransactionRequest),
    Eth(SignEthTransactionRequest),
}
```
will take either of these 2 JSONs, where type can be only Utxo or Eth
```json
{
  "coin": "KMD",
  "type": "Utxo",
  "tx": {
    "tx_hex": "",
    "prev_txns": [
    ]
  }
}
```
```json
{
  "coin": "ETH",
  "type": "Eth",
  "tx": {
    "value": "",
    "to": "",
    "data": ["", "", ""],
    "gas_limit": ""
  }
}
```

Reply:
I guess we could figure out 'type' from the coin object?

Reply:
fixed